.github/workflows/build.yml

```diff
@@ -59,8 +59,14 @@ jobs:
         with:
           distribution: 'zulu'
           java-version: ${{ matrix.java }}
+          cache: 'maven'
       - name: Build with Maven
         run: mvn -Pbuild-without-spotbugs -B package --file pom.xml
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@d9f34f8cd5cb3b3eb79b3e4b5dae3a16df499a70 # 3.1.1
+        if: ${{ matrix.java == '11.0.x' }} # publish results once
```

Comment:
This is good as long as we have a 1D matrix only. We could also use [`jobs.<job_id>.needs`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions) to structure this more explicitly as a fan-in, but I think this would make the build slower because that step would then have to rebuild the codebase and re-run the tests to get coverage. 

I think we should keep it this way because end-to-end build time is important, we just have to remember that if we ever want to do builds along another axis we're going to have to restructure this

Reply:
I'm not an expert in GitHub actions. Happy to replace this by something else better. What I wanted is to ensure we don't try to publish codecov results in all builds.

Reply:
I think the way it is now is best because if we make it more "general" it'll be slower (re-run build and tests) for no benefit at this point because we only matrix on one axis !